src/test/java/seedu/rc4hdb/logic/parser/AddCommandParserTest.java

```diff
@@ -1,142 +1,221 @@
+        Resident expectedResident = new ResidentBuilder(BOB).withTags(VALID_TAG_FRIEND).build();
 
         // whitespace only preamble
         assertParseSuccess(parser, PREAMBLE_WHITESPACE + NAME_DESC_BOB + PHONE_DESC_BOB + EMAIL_DESC_BOB
-                + ADDRESS_DESC_BOB + TAG_DESC_FRIEND, new AddCommand(expectedPerson));
+                + ROOM_DESC_BOB + GENDER_DESC_BOB + HOUSE_DESC_BOB + MATRIC_NUMBER_DESC_BOB + TAG_DESC_FRIEND,
+                new AddCommand(expectedResident));
 
         // multiple names - last name accepted
         assertParseSuccess(parser, NAME_DESC_AMY + NAME_DESC_BOB + PHONE_DESC_BOB + EMAIL_DESC_BOB
-                + ADDRESS_DESC_BOB + TAG_DESC_FRIEND, new AddCommand(expectedPerson));
+                + ROOM_DESC_BOB + GENDER_DESC_BOB + HOUSE_DESC_BOB + MATRIC_NUMBER_DESC_BOB + TAG_DESC_FRIEND,
+                new AddCommand(expectedResident));
 
         // multiple phones - last phone accepted
         assertParseSuccess(parser, NAME_DESC_BOB + PHONE_DESC_AMY + PHONE_DESC_BOB + EMAIL_DESC_BOB
-                + ADDRESS_DESC_BOB + TAG_DESC_FRIEND, new AddCommand(expectedPerson));
+                + ROOM_DESC_BOB + GENDER_DESC_BOB + HOUSE_DESC_BOB + MATRIC_NUMBER_DESC_BOB + TAG_DESC_FRIEND,
+                new AddCommand(expectedResident));
 
         // multiple emails - last email accepted
         assertParseSuccess(parser, NAME_DESC_BOB + PHONE_DESC_BOB + EMAIL_DESC_AMY + EMAIL_DESC_BOB
-                + ADDRESS_DESC_BOB + TAG_DESC_FRIEND, new AddCommand(expectedPerson));
+                + ROOM_DESC_BOB + GENDER_DESC_BOB + HOUSE_DESC_BOB + MATRIC_NUMBER_DESC_BOB + TAG_DESC_FRIEND,
+                new AddCommand(expectedResident));
 
-        // multiple addresses - last address accepted
-        assertParseSuccess(parser, NAME_DESC_BOB + PHONE_DESC_BOB + EMAIL_DESC_BOB + ADDRESS_DESC_AMY
-                + ADDRESS_DESC_BOB + TAG_DESC_FRIEND, new AddCommand(expectedPerson));
+        // multiple rooms - last room accepted
+        assertParseSuccess(parser, NAME_DESC_BOB + PHONE_DESC_BOB + EMAIL_DESC_BOB + ROOM_DESC_AMY
+                + ROOM_DESC_BOB + GENDER_DESC_BOB + HOUSE_DESC_BOB + MATRIC_NUMBER_DESC_BOB + TAG_DESC_FRIEND,
+                new AddCommand(expectedResident));
+
+        // multiple gender - last room accepted
+        assertParseSuccess(parser, NAME_DESC_BOB + PHONE_DESC_BOB + EMAIL_DESC_BOB + ROOM_DESC_BOB
+                + GENDER_DESC_AMY + GENDER_DESC_BOB + HOUSE_DESC_BOB + MATRIC_NUMBER_DESC_BOB + TAG_DESC_FRIEND,
+                new AddCommand(expectedResident));
+
+        // multiple houses - last room accepted
+        assertParseSuccess(parser, NAME_DESC_BOB + PHONE_DESC_BOB + EMAIL_DESC_BOB + ROOM_DESC_BOB
+                + GENDER_DESC_BOB + HOUSE_DESC_AMY + HOUSE_DESC_BOB + MATRIC_NUMBER_DESC_BOB + TAG_DESC_FRIEND,
+                new AddCommand(expectedResident));
+
+        // multiple matric numbers - last room accepted
```

Comment:
Some issues with the comments here. Might want to update "last room accepted" to reflect the corresponding field. 😄

Reply:
oh right hahaha